ksqldb-rest-app/src/test/java/io/confluent/ksql/api/integration/ApiIntegrationTest.java

```diff
@@ -659,6 +678,233 @@ private void shouldFailToExecuteQuery(final String sql, final String message) {
+    assertThat(printResponse.rows.get(1), containsString("rowtime:"));
+    assertThat(printResponse.rows.get(1), containsString(
+        "key: {\"F1\":[\"b\"]}, value: {\"STR\":\"BAR\",\"LONG\":2,\"DEC\":2.22,\"BYTES_\":\"Ag==\",\"ARRAY\":[],\"MAP\":{},\"STRUCT\":{\"F1\":3},\"COMPLEX\":{\"DECIMAL\":1.0,\"STRUCT\":{\"F1\":\"v1\",\"F2\":1},\"ARRAY_ARRAY\":[[\"foo\"]],\"ARRAY_STRUCT\":[{\"F1\":\"v1\"}],\"ARRAY_MAP\":[{\"k1\":1}],\"MAP_ARRAY\":{\"k\":[\"v1\"]},\"MAP_MAP\":{\"k\":{\"k\":1}},\"MAP_STRUCT\":{\"k\":{\"F1\":\"v1\"}}},\"TIMESTAMP\":2,\"DATE\":2,\"TIME\":1}, partition: 0"));
+    assertThat(printResponse.rows.get(2), containsString("rowtime:"));
+    assertThat(printResponse.rows.get(2), containsString(
+        "key: {\"F1\":[\"c\"]}, value: {\"STR\":\"BAZ\",\"LONG\":3,\"DEC\":30.33,\"BYTES_\":\"Aw==\",\"ARRAY\":[\"b\"],\"MAP\":{},\"STRUCT\":{\"F1\":null},\"COMPLEX\":{\"DECIMAL\":2.0,\"STRUCT\":{\"F1\":\"v2\",\"F2\":2},\"ARRAY_ARRAY\":[[\"foo\"]],\"ARRAY_STRUCT\":[{\"F1\":\"v2\"}],\"ARRAY_MAP\":[{\"k1\":2}],\"MAP_ARRAY\":{\"k\":[\"v2\"]},\"MAP_MAP\":{\"k\":{\"k\":2}},\"MAP_STRUCT\":{\"k\":{\"F1\":\"v2\"}}},\"TIMESTAMP\":3,\"DATE\":3,\"TIME\":2}, partition: 0"));
+  }
+
+  @Test
+  public void shouldFailToPrintNonExistingTopic() {
+    // Given:
+    String sql = "PRINT `NON_EXISTING_TOPIC`;";
+
+    // Then:
+    shouldFailToExecutePrint(sql, "Topic does not exist: NON_EXISTING_TOPIC");
+  }
+
+  @Test
+  public void shouldFailToPrintWithInvalidSql() {
+    // Given:
+    String sql = "PRONT `NON_EXISTING_TOPIC`;";
+
+    // Then:
+    shouldFailToExecutePrint(sql,
+        "line 1:1: Syntax Error\nUnknown statement 'PRONT'\nDid you mean 'PRINT'?\nStatement: PRONT `NON_EXISTING_TOPIC`;");
+  }
+
+  // Print topic currently only supports delimited format. Sendung an "accept" header with
+  // "application/json" or any other format will result in a 406 (Not Acceptable) response.
+  @Test
+  public void shouldFailToPrintInJsonFormat() {
+    // Given:
+    String sql = "PRINT " + TEST_TOPIC + " FROM BEGINNING LIMIT 1;";
+
+    // Create a write stream to capture the incomplete response
+    ReceiveStream writeStream = new ReceiveStream(vertx);
+
+    // Make the request to stream a print
+    JsonObject printRequestBody = new JsonObject().put("sql", sql);
+    VertxCompletableFuture<HttpResponse<Void>> responseFuture = new VertxCompletableFuture<>();
+
+    // When:
+    client.post("/query-stream").as(BodyCodec.pipe(writeStream))
+        .putHeader("Accept", "application/json")
```

Comment:
Thanks for adding this test. What happens if no explicit content type header is passed? From the code it looks like we default to using JSON from the `/query-stream` endpoint, which means a 406 will be thrown. Can we have a test for this?

Reply:
We default to using the delimited format, that's the check QueryStreamHandler, line 298:

```
    if (!(DELIMITED_CONTENT_TYPE.equals(contentType)
        || (contentType == null && !queryCompatibilityMode))) 
```

We actually have tests for this: all the other print tests aren't setting the accept header :)